modules/AOW_WorkFlow/AOW_WorkFlow.php

```diff
@@ -755,8 +788,28 @@ public function check_valid_bean(SugarBean $bean)
         return true;
     }
 
+    /**
+     * @param SugarBean $bean
+     * @return bool
+     */
+    protected function hasAlreadyRunToday(SugarBean $bean)
+    {
+        global $db, $timedate;
```

Comment:
do not use superglobals or globals if it's not necessary, pass it as an (optional) dependency.